azure-iot-provisioning-devicesdk/azure/iot/provisioning/devicesdk/security/sk_security_client.py

```diff
@@ -0,0 +1,64 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""This module contains a client that is responsible for providing shared access tokens that will eventually establish
+ the authenticity of devices to Device Provisioning Service.
+"""
+from azure.iot.common.sastoken import SasToken
+
+
+class SymmetricKeySecurityClient(object):
+    """
+    A client that is responsible for providing shared access tokens that will eventually establish
+    the authenticity of devices to Device Provisioning Service.
+    """
+
+    def __init__(self, registration_id, symmetric_key, id_scope):
```

Comment:
>id_scope [](start = 55, length = 8)

Ah, here's the id_scope.  I'm not exactly sure where the user should pass this in, but I think it belongs with the provisioning host.  It comes from exactly the same portal page as the provisioning host IIRC.

Reply:
so the `id_scope` is also needed by this class to produce the URI for `SasToken`. Since the host was only needed for the registration client I took the host parameter out and clubbed these 3 together..I understand what you mean by that the host and id_scope are available on portal page and the other 2 are available in enrollment details. I am open to suggestions if you feel we need to separate these out.

---
In reply to: [271431656](https://github.com/Azure/azure-iot-sdk-python-preview/pull/59#discussion_r271431656) [](ancestors = 271431656)